test/utils/gql_nodes_cache_test.dart

```diff
@@ -0,0 +1,388 @@
+import 'dart:typed_data';
+
+import 'package:ardrive/utils/gql_nodes_cache.dart';
+import 'package:flutter_test/flutter_test.dart';
+import 'package:shared_preferences/shared_preferences.dart';
+import 'package:stash/stash_api.dart';
+import 'package:stash_memory/stash_memory.dart';
+import 'package:stash_shared_preferences/stash_shared_preferences.dart';
+
+void main() {
+  TestWidgetsFlutterBinding.ensureInitialized();
+
+  group('MetadataCache class', () {
+    const mockDriveId = 'mockDriveId';
+    const mockDriveId2 = 'mockDriveId2';
+
+    test('can be constructed out of a Cache Store', () async {
+      final CacheStore cacheStore = await newMemoryCacheStore();
+
+      expect(
+        await GQLNodesCache.fromCacheStore(cacheStore),
+        isInstanceOf<GQLNodesCache>(),
+      );
+    });
+
+    test('can take as many entries as the maxEntries limit', () async {
+      final metadataCache = await GQLNodesCache.fromCacheStore(
+        await newMemoryCacheStore(),
+        maxEntries: 10,
+      );
+
+      final mockData = generateMockData(10);
+
+      for (int i = 0; i < mockData.length; i++) {
+        await metadataCache.put(mockDriveId, mockData[i]);
+        expect(await metadataCache.get(mockDriveId, i), mockData[i]);
+      }
+
+      final keys = await metadataCache.keys;
+      expect(keys, [
+        'mockDriveId_0',
+        'mockDriveId_1',
+        'mockDriveId_2',
+        'mockDriveId_3',
+        'mockDriveId_4',
+        'mockDriveId_5',
+        'mockDriveId_6',
+        'mockDriveId_7',
+        'mockDriveId_8',
+        'mockDriveId_9',
+      ]);
+    });
+
+    test('size indicates the number of entries in the cache', () async {
+      final metadataCache = await GQLNodesCache.fromCacheStore(
+        await newMemoryCacheStore(),
+        maxEntries: 10,
+      );
+
+      final mockData = generateMockData(1)[0];
+
+      await metadataCache.put(mockDriveId, mockData);
+
+      expect(await metadataCache.size, 1);
+    });
+
+    test('isFull indicates whether the cache is at max capacity', () async {
+      final metadataCache = await GQLNodesCache.fromCacheStore(
+        await newMemoryCacheStore(),
+        maxEntries: 10,
+      );
+
+      expect(await metadataCache.isFull, false);
+
+      final mockData = generateMockData(10);
+
+      for (int i = 0; i < mockData.length; i++) {
+        final wasPut = await metadataCache.put(mockDriveId, mockData[i]);
+        expect(wasPut, true);
+      }
```

Comment:
```suggestion
      for (int i = 0; i < mockData.length; i++) {
        final wasPut = await metadataCache.put(mockDriveId, mockData[i]);
        if (i < mockData.length) {
        	expect(metadataCache.isFull, false);
        }
        expect(wasPut, true);
      }
```
I would add this `isFull` expect as well

Reply:
Applied - f461436e5d5f585b4cba58b7de53b045a4b60528